tmt/utils.py

```diff
@@ -1064,6 +1094,26 @@ def default_branch(repository, remote='origin'):
         return ref.read().strip().split('/')[-1]
 
 
+def parse_dotenv(path: Path) -> Dict[str, str]:
+    def split_func(line):
+        return line.split("=")
+
+    try:
+        return dict(map(split_func, shlex.split(open(path), comments=True)))
+    except ValueError:
+        raise ValueError(f"Can't extract variables from {path}. "
+                         f"Ensure it has proper format (i.e. A=B)")
+
+
+def parse_yaml(path: Path) -> Dict[str, str]:
+    yaml_as_dict = yaml.safe_load(path.read_text())
+    assert not any(isinstance(val, dict) for val in yaml_as_dict.values()), (
```

Comment:
Should raise exception (tmt.utils.GeneralError) which is processed nicely to the user.